views/ABViewCore.js

```diff
@@ -787,14 +790,23 @@ module.exports = class ABViewCore extends ABMLClass {
       // change id
       result.id = lookUpIds[result.id] || this.application.uuid();
 
+      // page's name should not be duplicate
+      if (this.key == "page") {
+         result.name = `${result.name}_copied_${this.application
+            .uuid()
+            .slice(0, 3)}`;
+
+         result.label = `${result.label} (copied)`;
+      }
+
       return Promise.resolve()
          .then(() => {
             // copy sub pages
             var allSaves = [];
 
             if (this._pages && !options.ignoreSubPages) {
                result._pages = [];
-               this.pages().forEach((p) => {
+               this.pages((p) => p && p.isRoot()).forEach((p) => {
```

Comment:
So in this case: if I have ```Page Top``` -> ```Page mid``` -> ```Page Bottom```  and I want to copy, ```Page mid```,  shouldn't the ```Page mid copy``` also have a ```Page Bottom copy``` under it?

With this check in place, it would prevent the ```Page mid copy``` from also having a copy of ```Page Bottom```.